src/main/java/se/liu/ida/hefquin/engine/queryplan/logical/LogicalPlanUtils.java

```diff
@@ -20,6 +22,55 @@ static public boolean isSourceAssignment( final LogicalPlan plan ) {
 		LogicalPlanWalker.walk(plan, null, v);
 		return v.wasSourceAssignment();
 	}
+	
+	static public int countSubplans ( final LogicalPlan plan ) {
+		final LogicalPlanCounter c = new LogicalPlanCounter();
+		LogicalPlanWalker.walk(plan, null, c);
+		return c.getSubplanCount();
+	}
+	
+	static public class LogicalPlanCounter extends LogicalPlanVisitorBase {
```

Comment:
```suggestion
	static public class LogicalPlanCounter implements LogicalPlanVisitor {
```
Better this way to make sure you don't miss overriding any of the `visit` methods.